lib/schema/Schemas.js

```diff
@@ -0,0 +1,396 @@
+const semverRegex = "^[0-9]+\.[0-9]+\.[0-9]+(-.+)?$"
+
+const pluginDefinitionSchema = Object.freeze({
```

Comment:
This has no notion of version. We're going to make improvements over time, new features, and so on. So, the pluginDefinition schema will change by version. Each plugin contains an attribute, `apiVersion`, which describes which schema it is attempting to conform to. So, what you have here is the schema for 1.0.0.
You need the server to be able to take multiple schemas, and test each plugin against the schema definition that matches the one it states it is using.